### PR TITLE
feat(client): ✨ Signin

### PR DESCRIPTION
**Pull request by zaverbac**

### CODE REVIEW

**Review by nstringham — CHANGES_REQUESTED** [review]
Can you please remove all the unused images

**Review by nstringham — CHANGES_REQUESTED** [review]
Please read over the linting errors and understand what they are saying and fix them if appropriate

**Comment by nstringham** [comment]
it might be a good idea to make a file for auth instead of putting it all in the firebase file

**Comment by zaverbac** [comment]
Completed all suggestions in review

File: client/src/lib/firebase/authfunctions.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+export async function loginWithGoogle() {
+  const auth = getAuth(app);
+  const provider = new GoogleAuthProvider();
+  signInWithPopup(auth, provider).then((result) => {
+    // This gives you a Google Access Token. You can use it to access the Google API.
+    const credential = GoogleAuthProvider.credentialFromResult(result);
+    const token = credential!.accessToken;
+    // The signed-in user info.
+    const user = result.user;
+  });
```
Comment (nstringham) [inline client/src/lib/firebase/authfunctions.ts]: it looks like we are not using any of these values so we probobly don't need the then block

File: client/src/lib/firebase/authfunctions.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  signInWithPopup,
+  signInAnonymously,
+  signInWithEmailAndPassword,
+} from "firebase/auth";
+
+// Google login/signup
+export async function loginWithGoogle() {
+  const auth = getAuth(app);
+  const provider = new GoogleAuthProvider();
+  signInWithPopup(auth, provider).then((result) => {
```
Comment (nstringham) [inline client/src/lib/firebase/authfunctions.ts]: `signInWithPopup` is asyncronous so it should be awaited or returned

File: client/src/lib/firebase/authfunctions.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    const user = result.user;
+  });
+}
+
+// Microsoft login/signup
+export async function loginWithMicrosoft() {
+  const auth = getAuth();
+  const provider = new OAuthProvider("microsoft.com");
+
+  signInWithPopup(auth, provider);
```
Comment (nstringham) [inline client/src/lib/firebase/authfunctions.ts]: `signInWithPopup` is asyncronous so it should be awaited or returned

File: client/src/lib/firebase/authfunctions.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+// Anonymous login
+export function loginAnonymously() {
+  const auth = getAuth();
+  signInAnonymously(auth);
```
Comment (nstringham) [inline client/src/lib/firebase/authfunctions.ts]: `signInAnonymously` is asyncronous so it should be awaited or returned

**Review by nstringham — CHANGES_REQUESTED** [review]

**Comment by nstringham** [comment]
related to #27

File: client/src/lib/stores/authstore.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+export function getID() {
+  const { subscribe, set, update } = writable(localStorage.getItem("name"));
+
+  return {
+    subscribe,
+  };
+}
+if (typeof localStorage !== "undefined") {
+  getID();
+}
```
Comment (nstringham) [inline client/src/lib/stores/authstore.ts]: I think we should remove this in favor of rxfire

**Review by nstringham — CHANGES_REQUESTED** [review]
Once you remove `authstore.ts` it should be good enough to merge and then we can make it better later

File: client/src/routes/signin/+page.svelte [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+      ". . . . . . . . ."
+      ". . . . . . . . ."
+      ". . . . . . . . ."
+      ". . . . google . . . ."
+      ". . . . microsoft . . . ."
+      ". . . . email . . . ."
+      ". . . . input . . . ."
+      ". . . . . . . . ."
+      ". . . . . . . . ."
+      ". . . . . . . . .";
```
Comment (nstringham) [inline client/src/routes/signin/+page.svelte:36]: If you are trying to center you should consider using `place-content`
https://css-tricks.com/snippets/css/complete-guide-grid/#aa-place-content

**Review by nstringham — APPROVED** [review]
Good enough for now

**Pull request merged**